cdap-docs/admin-manual/source/system-requirements.rst

```diff
@@ -176,3 +179,27 @@ NTP (Network Time Protocol)
 - If you need to adjust the configuration (add or delete servers, use servers closer to you, etc.)::
 
     $ vi /etc/ntp.conf
+
+
+.. _admin-manual-cdap-and-firewalls:
+
+CDAP and Firewalls
+==================
+In your :ref:`cluster configuration <admin-manual-install-deployment-architectures>`, 
+there cannot be a firewall between the cluster and CDAP. Instead, if a
```

Comment:
It's a little hard to tell from that picture, but... it shows the CDAP Router "traversing" the firewall. In other words, UI can be totally outside the firewall, as it needs to talk to Auth/Router. These two services need to be accessible from the outside (users) and also be able to connect to anything within the cluster. They need unrestricted *client* access to the cluster... the ability to establish connections to the cluster nodes, on any port that a container may open.

Taking that same picture, move the firewall *just* left of Auth/Router, and it needs two holes allowed through from clients, 10000 and 10009... the hosts running Auth/Router need to be able to establish connections. There could easily be another firewall between Auth and the cluster, so long as it provided client access from Auth to the ZooKeeper nodes. The same is true for Router, except it needs client access to pretty much everything, so it doesn't make much sense to firewall it when you're essentially allowing all traffic through.

This is *mostly* true, in practice. However, we should really document all the client connections these services can make as some organizations have policies against open firewalls for any hosts.